maventa/auth.ts

```diff
@@ -0,0 +1,52 @@
+import https from 'https';
+import querystring from 'querystring';
+import config from './config';
+
+interface TokenResponse {
+  access_token: string;
+  token_type: string;
+  expires_in: number;
+  scope: string;
+}
+
+export async function getAccessToken(): Promise<string> {
+  const postData = querystring.stringify({
+    grant_type: 'client_credentials',
+    client_id: config.clientId,
+    client_secret: config.clientSecret,
+    scope: config.scope,
+    vendor_api_key: config.vendorApiKey,
+  });
+
+  const options = {
+    hostname: config.baseUrl,
+    path: '/oauth2/token',
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+      'Content-Length': postData.length,
+    },
+  };
+
+  return new Promise<string>((resolve, reject) => {
+    const req = https.request(options, (res) => {
```

Comment:
This should use HttpService or directly our Request implementation. Not directly platform specific NodeJS depencency.

Reply:
See https://github.com/hyperifyio/io.hyperify.core/?tab=readme-ov-file#request